src/avisi/rsync/local_dir.clj

```diff
@@ -59,17 +66,21 @@
 (defn- path->file-details [root-path file]
   (let [absolute-path (.getAbsolutePath file)
         rel-path (relative-path root-path absolute-path)
-        md5 (md5/md5-file absolute-path)]
-    {:path rel-path :md5 md5}))
+        md5 (md5/md5-file absolute-path)
+        last-modified (.lastModified file)
+        size (.length file)]
+    {:path rel-path :md5 md5 :meta {:size size :timestamp last-modified}}))
 
 (defrecord DirectoryLocation [directory]
   l/Location
   (analyse [this]
     (analyse-local-directory directory))
   (write [this path stream]
-    (write (str directory "/" path) stream))
+    (write (str directory "/" (:path path)) stream))
```

Comment:
This are probably the file details maybe destructure immediately like this: `{:keys [path]}`